3.0.0-beta.0-Button-Fab-Styles

onluiz opened the pull request:
# Description

Summary: Change FAB button so it can receive custom styles in its declaration.
Motivation: Now we can customize FAB and change its position on screen (not only on bottom right)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- unit testing
- browser testing

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

onluiz commented on src/Button.js:
```diff
@@ -89,8 +89,10 @@ class Button extends Component {
+    const { fab, floating, large, ...other } = this.props;
     return (
       <div
+        {...other}
```
No :)
But I will remove className from {...other} either way

alextrastero replied:
both classname props are merged? or how does this work?

onluiz replied:
No :) only the last attribute ends up into the div.

But it does not sound quite a good practice, so I will make sure only the correct attribute is passed to the div.

Thanks for the feedback @alextrastero!

onluiz commented on test/Button.spec.js:
```diff
@@ -177,5 +168,11 @@ describe('Button', () => {
+      const style = { bottom: '45px', right: '24px' };
+      wrapper = mount(FabButton(true, style));
+      expect(wrapper.props().style.bottom).toEqual(style.bottom);
+      expect(wrapper.props().style.right).toEqual(style.right);
```
Done! :)
But I wanted to check a detail with you. The prop style is always defined, even when I pass undefined to it. So for that matter I'm checking if the defined object has attributes.

It dind't felt right, so I wanted your opinion about it. :)

This change is in here: https://github.com/react-materialize/react-materialize/pull/741/commits/9233d684f91a95ae4f4e3359e8814626c07c3936#diff-57f49d39d70d58986a35e5a8d4f5fbf6R171

Thank you @alextrastero!

alextrastero commented on src/Button.js:
```diff
@@ -88,13 +88,15 @@ class Button extends Component {
 
-  renderFab(className) {
+  renderFab(classes) {
+    const { fab, floating, large, className, ...other } = this.props;
```
aren't you getting an unused prop variable warning?

alextrastero replied:
on L73 we are passing 2 params to `renderFab` but only using one :)

onluiz commented:
> Also @onluiz I've just realised that materialize-css's FAB doesn't have the ability to change the FAB's position. Is this something we really need? Couldn't the `fixed-action-btn` be wrapped in another div absolute positioned to achieve the wanted behaviour?

I tried something like that, but it looks like `fixed-action-btn` is position fixed:
```
.fixed-action-btn {
    position: fixed;
    right: 23px;
    bottom: 23px;
    padding-top: 15px;
    margin-bottom: 0;
    z-index: 997;
}
```
Is there a way to change this into another div? If so, as you said, that'd be a better solution.

Thanks for the feedback! :)

alextrastero commented:
The thing is, we don't want to have features that materialize-css doesn't include, we are just a react wrapper, if materialize-css support this in any way then i'm up for the change.

onluiz commented:
> The thing is, we don't want to have features that materialize-css doesn't include, we are just a react wrapper, if materialize-css support this in any way then i'm up for the change.

I see. But it is ok to our component not to be able of accepting style attribute?

alextrastero commented:
True, the component should be allowed to receive `other` props 👍
